hugr-core/src/hugr/validate.rs

```diff
@@ -834,5 +815,15 @@ pub enum InterGraphEdgeError {
     },
 }
 
+#[derive(Debug, Clone, PartialEq, Error)]
+#[error("Extensions of child node ({child}) {child_extensions} are not a subset of the parent node ({parent}): {parent_extensions}")]
+/// An error in the extension deltas.
+pub struct ExtensionError {
+    parent: Node,
+    parent_extensions: ExtensionSet,
+    child: Node,
+    child_extensions: ExtensionSet,
+}
+
```

Comment:
The solution is always, add it to the parent delta...ok, I've changed the string a bit, how's that?

I think this is basically the only error. It feels generic because we are used to a wider variety of errors with the current inference scheme, but these basically all go away. I say "basically" because it's possible that we might want a slightly different message when we start doing inference again (of deltas), but even then there's still only really one possibility i.e. that "the parent annotation is too restrictive".